vm/vm_test.go

```diff
@@ -778,41 +780,69 @@ func runInterruptTest(t *testing.T, script string) {
 		t.Errorf("Define error: %v", err)
 	}
 
+	ctx, cancel := context.WithCancel(context.Background())
 	go func() {
 		<-waitChan
-		Interrupt(env)
+		cancel()
 	}()
 
-	_, err = env.Execute(script)
-	if err == nil || err.Error() != ErrInterrupt.Error() {
+	_, err = env.ExecuteContext(script, ctx)
+	if err == nil || !strings.Contains(err.Error(), ErrInterrupt.Error()) {
```

Comment:
This one is in strings.Contains because the `run` [function](https://github.com/mattn/anko/blob/ad2f2da49381990329902ee5a96f0833ad1cb0e1/vm/vmStmt.go#L29) is [recursive](https://github.com/mattn/anko/blob/ad2f2da49381990329902ee5a96f0833ad1cb0e1/vm/vmStmt.go#L209), and [wrap](https://github.com/mattn/anko/blob/ad2f2da49381990329902ee5a96f0833ad1cb0e1/vm/vmStmt.go#L86) the returning errors.
Which means that the ErrInterrupt error that you receive is [sometime wrapped](https://github.com/mattn/anko/blob/ad2f2da49381990329902ee5a96f0833ad1cb0e1/vm/vmExpr.go#L723), therefore cannot be check with `==`.

Reply:
That might be fixed once the other changes have been done. If not, we probably need look into it and fix that issue.

Reply:
Nevermind, I think it was failing when I was trying to compare the error itself (not their message).
The original error checking works just fine.